pkg/ottl/contexts/ottlprofile/profile.go

```diff
@@ -0,0 +1,236 @@
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal"
+)
+
+const (
+	// Experimental: *NOTE* this constant is subject to change or removal in the future.
+	ContextName            = "profile"
+	contextNameDescription = "Profile"
+)
+
+var (
+	_ internal.ResourceContext             = (*TransformContext)(nil)
+	_ internal.InstrumentationScopeContext = (*TransformContext)(nil)
+	_ internal.ProfileContext              = TransformContext{}
+
+	// _ zapcore.ObjectMarshaler              = (*TransformContext)(nil)
+)
+
+type TransformContext struct {
+	profile              pprofile.Profile
+	instrumentationScope pcommon.InstrumentationScope
+	resource             pcommon.Resource
+	cache                pcommon.Map
+	scopeProfiles        pprofile.ScopeProfiles
+	resourceProfiles     pprofile.ResourceProfiles
+}
+
+type Option func(*ottl.Parser[TransformContext])
+
+type TransformContextOption func(*TransformContext)
+
+func NewTransformContext(profile pprofile.Profile, instrumentationScope pcommon.InstrumentationScope, resource pcommon.Resource, scopeProfiles pprofile.ScopeProfiles, resourceProfiles pprofile.ResourceProfiles, options ...TransformContextOption) TransformContext {
+	tc := TransformContext{
+		profile:              profile,
+		instrumentationScope: instrumentationScope,
+		resource:             resource,
+		cache:                pcommon.NewMap(),
+		scopeProfiles:        scopeProfiles,
+		resourceProfiles:     resourceProfiles,
+	}
+	for _, opt := range options {
+		opt(&tc)
+	}
+	return tc
+}
+
+// Experimental: *NOTE* this option is subject to change or removal in the future.
+func WithCache(cache *pcommon.Map) TransformContextOption {
+	return func(p *TransformContext) {
+		if cache != nil {
+			p.cache = *cache
+		}
+	}
+}
+
+func (tCtx TransformContext) GetProfile() pprofile.Profile {
+	return tCtx.profile
+}
+
+func (tCtx TransformContext) GetInstrumentationScope() pcommon.InstrumentationScope {
+	return tCtx.instrumentationScope
+}
+
+func (tCtx TransformContext) GetResource() pcommon.Resource {
+	return tCtx.resource
+}
+
+func (tCtx TransformContext) getCache() pcommon.Map {
+	return tCtx.cache
+}
+
+func (tCtx TransformContext) GetScopeSchemaURLItem() internal.SchemaURLItem {
+	return tCtx.scopeProfiles
+}
+
+func (tCtx TransformContext) GetResourceSchemaURLItem() internal.SchemaURLItem {
+	return tCtx.resourceProfiles
+}
+
+func NewParser(functions map[string]ottl.Factory[TransformContext], telemetrySettings component.TelemetrySettings, options ...Option) (ottl.Parser[TransformContext], error) {
+	pep := pathExpressionParser{telemetrySettings}
+	p, err := ottl.NewParser[TransformContext](
+		functions,
+		pep.parsePath,
+		telemetrySettings,
+		ottl.WithEnumParser[TransformContext](parseEnum),
```

Comment:
Since there are no enums in otlp Profiles we can we can remove this option and the `parseEnum` function.